### PR TITLE
zephyr: Add SM/MAS/OOB/BV-01-C with use of gap_oob_legacy_set_data

### PR DESCRIPTION
**Pull request by AyturkDuzen**

### CODE REVIEW

File: ptsprojects/zephyr/sm.py [hunk 1]
```diff
@@ -115,6 +115,7 @@ def test_cases(pts):
     pts_bd_addr = pts.q_bd_addr
 
     stack = get_stack()
+    stack.gap_init()
```
Comment (kromancer) [inline ptsprojects/zephyr/sm.py]: I see that `stack.gap_init()` is part of the `pre_conditions`.
Is there any reason we want to call it here/twice?

Comment (ryanjh) [reply ptsprojects/zephyr/sm.py]: The same question to me. The others look good.

**Review by ryanjh — CHANGES_REQUESTED** [review]
Reviewed.

File: ptsprojects/zephyr/sm.py (hunk 1 above)
Comment (AyturkDuzen) [reply ptsprojects/zephyr/sm.py]: stack.gap.oob_legacy is called before pre_conditions which sets gap.init. If I do not set it there, we get the following:
AttributeError: 'NoneType' object has no attribute 'oob_legacy'

From what I see from mynewt, they initialize gap as the main function and don't put in preconditions. We can do a similar trick.

**Review by kromancer — APPROVED** [review]

**Review by ryanjh — APPROVED** [review]

**Pull request merged**